mitiq/executor/executor.py

```diff
@@ -36,6 +37,8 @@
     Sequence[np.ndarray],  # type: ignore
     Tuple[np.ndarray],
     npt.NDArray[np.complex64],
+    list[npt.NDArray[np.complex64]],
```

Comment:
We could align the rest of types to have the same more specific parameters `npt.NDArray[np.complex64]` instead of `np.ndarray`

Reply:
This is tightly linked with my comment in https://github.com/unitaryfund/mitiq/pull/2700#discussion_r2008809787 about how we actually use these types (i.e. how we use member inclusion instead of subclassing). I've tried to replace the `np.ndarray` generics but then tests start failing because `np.ndarray` (a potential typehint for a density matrix executor) fails to be in `DensityMatrixLike`.

I'm not sure what the best solution here is because someone could use `np.ndarray` as a typehint, or `npt.NDArray[...]`. How far do we go to support every typehint?